docs/gc.md

```diff
@@ -141,15 +141,20 @@ Object ages are tracked for each region with a maximum of 24 possible generation
 
-Over time, transient objects drop out of the live set and viable survivor regions become scarce. This problem is solved by introducing a global mark phase (GMP) to mark and sweep the entire heap. GMP is a hybrid *incremental-concurrent* GC cycle. Each GMP increment is interleaved with a PGC cycle so that regions that are left sparse after sweeping the heap can be collected, cleared, and recycled during and after the GMP.
+Partial GC cycles work to reclaim free regions in the heap for allocating new objects. As much as 90% of this memory is reclaimed by moving objects out of *eden* regions to older regions.
+However, because some objects from *eden* regions always survive, the partial GC cycle must also work to reclaim free regions by defragmenting older regions. For example, a partial GC cycle that moves objects from 5 fragmented older regions into 2 empty regions, reclaims 3 regions for new object allocation. However, over time the overall amount of fragmented memory decreases and records about object liveness in older regions becomes less accurate. Eventually, the work done by partial GC cycles to reclaim memory cannot keep pace with memory allocation. Free regions become so scarce that a global *mark* operation (GMP) is required to build a new record of object liveness across the heap. A *sweep* operation uses this record to measure the amount of free memory in fragmented older regions, which later partial GC cycles can act upon to move objects and reclaim free regions. 
 
-With the `balanced` policy, a global GC cycle is rare, occurring only in very tight memory conditions when PGC and GMP operations cannot free enough memory on the heap. These global GCs are explicit GCs, like those invoked by an application calling `System.gc()`.
+A global *sweep* operation also runs to reclaim memory. This operation creates further fragmented memory that partial GC cycles can work on. The global *sweep* operation, while logically associated with the global *mark* operation, runs in the first partial GC cycle after the *mark* operation completes. Because the GC cycle responsible for the global *mark* operation runs concurrently, it might overlap and interleave with a few partial GC cycles.
```

Comment:
Consider expaning this sentence:

A global *sweep* operation also runs to reclaim memory, so that it create completely empty regions.

This sentence 

This operation creates further fragmented memory that partial GC cycles can work on.

is actually more similar to 

A *sweep* operation uses this record to measure the amount of free memory in fragmented older regions, which later partial GC cycles can act upon to move objects and reclaim free regions. 
 
and it can be omitted.